src/sources/kubernetes_logs/mod.rs

```diff
@@ -278,6 +285,20 @@ pub struct Config {
     #[configurable(metadata(docs::type_unit = "seconds"))]
     #[serde(default = "default_rotate_wait", rename = "rotate_wait_secs")]
     rotate_wait: Duration,
+
+    /// The strategy to use for log collection.
+    log_collection_strategy: LogCollectionStrategy,
+}
+
+/// Configuration for the log collection strategy.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+enum LogCollectionStrategy {
+    /// Collect logs by reading log files from the filesystem.
+    File,
+    /// Collect logs via the Kubernetes Logs API.
+    Api,
 }
```

Comment:
This needs a default value in order not to break existing users. I **think** this works:


```suggestion
    /// The strategy to use for log collection.
    #[serde(default)]
    log_collection_strategy: LogCollectionStrategy,
}

/// Configuration for the log collection strategy.
#[configurable_component]
#[derive(Clone, Copy, Debug, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
enum LogCollectionStrategy {
    /// Collect logs by reading log files from the filesystem.
    #[default]
    File,
    /// Collect logs via the Kubernetes Logs API.
    Api,
}
```

You also need to run `make generate-component-docs` in order to generate documentation for this and then we can see if it all looks ok there too